test/integration/controller/core/clusterqueue_controller_test.go

```diff
@@ -190,4 +193,117 @@ var _ = ginkgo.Describe("ClusterQueue controller", func() {
 			}))
 		})
 	})
+
+	ginkgo.When("Creating clusterQueues", func() {
+		var cq *kueue.ClusterQueue
+		ginkgo.BeforeEach(func() {
+			cq = testing.MakeClusterQueue("foo-cq").Obj()
+			gomega.Expect(k8sClient.Create(ctx, cq)).To(gomega.Succeed())
+		})
+		ginkgo.AfterEach(func() {
+			gomega.Expect(framework.DeleteClusterQueue(ctx, k8sClient, cq)).To(gomega.Succeed())
+		})
+
+		ginkgo.It("Should add a finalizer to the clusterQueue", func() {
+			gomega.Eventually(func() []string {
+				var newCQ kueue.ClusterQueue
+				gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(cq), &newCQ)).To(gomega.Succeed())
+				return newCQ.GetFinalizers()
+			}, framework.Timeout, framework.Interval).Should(gomega.Equal([]string{kueue.ResourceInUseFinalizerName}))
+		})
+	})
+
+	ginkgo.When("Deleting clusterQueues", func() {
+		var (
+			cq    *kueue.ClusterQueue
+			queue *kueue.Queue
+		)
+
+		ginkgo.BeforeEach(func() {
+			cq = testing.MakeClusterQueue("foo-cq").Obj()
+			queue = testing.MakeQueue("queue", ns.Name).ClusterQueue(cq.Name).Obj()
+			gomega.Expect(k8sClient.Create(ctx, queue)).To(gomega.Succeed())
+			gomega.Expect(k8sClient.Create(ctx, cq)).To(gomega.Succeed())
+		})
+
+		ginkgo.It("Should delete clusterQueues successfully when no admitted workloads are running", func() {
+			gomega.Expect(framework.DeleteClusterQueue(ctx, k8sClient, cq)).To(gomega.Succeed())
+			gomega.Eventually(func() error {
+				var newCQ kueue.ClusterQueue
+				return k8sClient.Get(ctx, client.ObjectKeyFromObject(cq), &newCQ)
+			}, framework.Timeout, framework.Interval).Should(testing.BeNotFoundError())
+		})
+
+		ginkgo.It("Should stuck in termination until admitted workloads finished running", func() {
+			ginkgo.By("Admit workloads")
+			admission := testing.MakeAdmission(cq.Name).Obj()
+			wl := testing.MakeWorkload("workload", ns.Name).Queue(queue.Name).Admit(admission).Obj()
+			gomega.Expect(k8sClient.Create(ctx, wl)).To(gomega.Succeed())
+
+			ginkgo.By("Delete clusterQueue")
+			gomega.Expect(framework.DeleteClusterQueue(ctx, k8sClient, cq)).To(gomega.Succeed())
+			var newCQ kueue.ClusterQueue
+			gomega.Eventually(func() []string {
+				gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(cq), &newCQ)).To(gomega.Succeed())
+				return newCQ.GetFinalizers()
+			}, framework.Timeout, framework.Interval).Should(gomega.Equal([]string{kueue.ResourceInUseFinalizerName}))
+
+			ginkgo.By("New created workloads should not be admitted")
+			gomega.Eventually(func() *kueue.Admission {
+				newWL := testing.MakeWorkload("new-workload", ns.Name).Queue(queue.Name).Obj()
+				gomega.Expect(k8sClient.Create(ctx, newWL)).To(gomega.Succeed())
+				return newWL.Spec.Admission
+			}, framework.Timeout, framework.Interval).Should(gomega.BeNil())
```

Comment:
instead we need to check that the number of admitted workloads in CQ status is `1`; this will test that reconcile was invoked, but the finalizer wasn't removed yet.